python-package/xgboost/data.py

```diff
@@ -1,553 +1,660 @@
-# pylint: disable=too-many-arguments, no-self-use, too-many-instance-attributes
+# pylint: disable=too-many-arguments, too-many-branches
+# pylint: disable=too-many-return-statements, import-error
 '''Data dispatching for DMatrix.'''
 import ctypes
-import abc
 import json
 import warnings
 
 import numpy as np
 
-from .core import c_array, _LIB, _check_call, c_str, _cudf_array_interfaces
-from .core import DataIter
-from .compat import lazy_isinstance, STRING_TYPES, os_fspath, os_PathLike
+from .core import c_array, _LIB, _check_call, c_str
+from .core import DataIter, DeviceQuantileDMatrix, DMatrix
+from .compat import lazy_isinstance, os_fspath, os_PathLike
 
 c_bst_ulong = ctypes.c_uint64   # pylint: disable=invalid-name
 
 
-class DataHandler(abc.ABC):
```

Comment:
So we are completely abandoning `DataHandler` class?

Reply:
Yes.  Just plain old function dispatching.  Not clean, but at least debugger friendly ...